src/graph.rs

```diff
@@ -95,14 +94,9 @@ impl<'tree> Graph<'tree> {
 
     pub fn display_json<'a>(&'a self, ctx: &'a Context) -> () {
         // TODO: move this all into a new module
-        struct JSONGraph<'a, 'tree>(&'a Graph<'tree>, &'a Context);
-        struct JSONNode<'a>(usize, &'a GraphNode, &'a Context);
-        struct JSONEdge<'a>(&'a u32, &'a Edge, &'a Context);
-        struct JSONAttributes<'a>(&'a Attributes, &'a Context);
-        struct JSONIdentifier<'a>(&'a Identifier, &'a Context);
-        struct JSONValue<'a>(&'a Value, &'a Context);
-
-        impl<'a, 'tree> ser::Serialize for JSONGraph<'a, 'tree> {
+        struct InContext<'a, T>(&'a T, &'a Context);
```

Comment:
This works out really nicely. I should use that in other places...

Reply:
Yeah, I was quite pleased with it.